lib/Core/SQLiteBuildDB.cpp

```diff
@@ -598,15 +602,26 @@ class SQLiteBuildDB : public BuildDB {
     return true;
   }
 
-  virtual void buildStarted() override {
+  virtual bool buildStarted(std::string *error_out) override {
     // Execute the entire build inside a single transaction.
     //
     // FIXME: We should revist this, as we probably wouldn't want a crash in the
     // build system to totally lose all build results.
-    int result = sqlite3_exec(db, "BEGIN IMMEDIATE;", nullptr, nullptr,
-        nullptr);
-    assert(result == SQLITE_OK);
-    (void)result;
+    char *cError;
+    int result = sqlite3_exec(db, "PRAGMA locking_mode = EXCLUSIVE", nullptr, nullptr, &cError);
```

Comment:
Based on my reading of:
 https://sqlite.org/pragma.html
we shouldn't actually need this -- I think it would be ok to only have an EXCLUSIVE lock on the db for the duration of the build.